### PR TITLE
add model parser

### PR DESCRIPTION
**Pull request by y-f-u**
This adds a serde parse function to parse `models`.

Which can be:
```yaml
models:
- from: spice.ai/xyz/model-a
  name: model-a
  datasets:
    - eth_blocks
```

### CODE REVIEW

File: crates/runtime/src/modelformat.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pub mod onnx;
+
+pub trait ModelFormat {
+    fn from_bytes(bytes: &[u8]) -> Self;
```
Comment (y-f-u) [inline crates/runtime/src/modelformat.rs:4]: this is a placeholder for the format implementation later

File: crates/spicepod/src/component.rs [hunk 2]
```diff
@@ -16,7 +17,7 @@ pub struct ComponentReference {
     pub r#ref: String,
 
     #[serde(skip_serializing_if = "Vec::is_empty")]
-    #[serde(rename = "dependsOn", default)]
+    #[serde(rename = "dependsOn", alias = "datasets", default)]
```
Comment (y-f-u) [inline crates/spicepod/src/component.rs:20]: reuse component reference for model too.

Comment (phillipleblanc) [reply crates/spicepod/src/component.rs:20]: `dependsOn` is a different concept - we shouldn't use it to represent what datasets the model depends on

Comment (phillipleblanc) [reply crates/spicepod/src/component.rs:20]: Actually maybe this is fine.

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**